### PR TITLE
geckolib: Add style structs bindings and documentation

### PR DESCRIPTION
**Pull request by emilio**
See the description of 6c851c8 to see which incorrect bindings remain. They're just nine types.

cc @bholley @SimonSapin @pcwalton 

(I don't know who should review this actually)

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10194)

### CODE REVIEW

**Comment by emilio** [comment]
Relevant Gecko patch: https://bugzilla.mozilla.org/show_bug.cgi?id=1259802

File: ports/geckolib/gecko_style_structs.rs [hunk 1]
```diff
@@ -0,0 +1,891 @@
+/* automatically generated by rust-bindgen */
```
Comment (bholley) [inline ports/geckolib/gecko_style_structs.rs]: This is awesome! Way better than the 1.4 megabytes of generated code that I came up with. ;-)

File: ports/geckolib/tools/regen_style_structs.sh [hunk 2]
```diff
@@ -50,16 +50,40 @@ fi
...
+  -match "nsTSubstring"                                             \
+  -match "nsTString"                                                \
+  -blacklist-type "nsStringComparator"                              \
+  -blacklist-type "nsDefaultStringComparator"                       \
   -include "$1/mozilla-config.h"                                    \
   -o ../gecko_style_structs.rs                                      \
-  "$DIST_INCLUDE/nsStyleStruct.h"
+  "$DIST_INCLUDE/nsString.h"
+yes \
+  -match "nsStyleStruct"                                            \
```
Comment (bholley) [inline ports/geckolib/tools/regen_style_structs.sh]: It might be clearer to put .h after these to make it obvious these are headers, not types.

Comment (emilio) [inline ports/geckolib/tools/regen_style_structs.sh]: This one for example matches both nsStyleStruct and nsStyleStructFwd.h, but I agree.

I think in the last patch I include quite a few `.h`s though.

**Pull request merged**